### ISSUE TITLE
How to easily setup local dev instance?

### ISSUE THREAD

**Issue by fooness** [mastodon/mastodon#10283]
Is there any way to easily setup a local development instance?

Think there was a vagrant workflow once, but it’s not in the docs anymore …

**Comment by Gargron** [comment]
Isn't it just `git clone ... & cd ... & vagrant up`? There is a Vagrantfile in the repository. What documentation would you like to see for it?

**Comment by fooness** [comment]
That sounds easy indeed. Is it mentioned somewhere in the docs or in the repo? Maybe I’ve overseen something. For people being aware of vagrant it’s probably enough to see a Vagrantfile.

Seems there’s an ongoing bug in vagrant which sometimes(?) prevents mounting NFS shares. https://github.com/hashicorp/vagrant/issues/5802

**Comment by Grawl** [comment]
I found Vagrant guide here: https://framagit.org/josephk/framapiaf/blob/master/docs/Running-Mastodon/Vagrant-guide.md

But not in this repository or official documentation.

I think it will be good to explain how I can use to bootstrap Mastodon on server and locally. There is a lot of files in repository root, and `Dockerfile` with `Vagrantfile` should be mentioned in `readme` and explained with links to guides in docs.

In official documentation, I found [Setting up a dev environment](https://docs.joinmastodon.org/dev/setup/) and it suggests me to use local software to set up. I prefer virtualization using Docker. Vagrant is ok too for me.

Finally, I run `vagrant up` and see `Welcome to nginx!` because of #2749 but Node is at version 8 instead of 10.
